### PR TITLE
[WFCORE-4332] Add the named-formatter attribute to the syslog-handler.

### PR DESCRIPTION
**Pull request by jamezp**
JIRA for schema bump: https://issues.jboss.org/browse/WFCORE-4337

JIRA for `named-formatter` addition: https://issues.jboss.org/browse/WFCORE-4332

### CODE REVIEW

**Comment by jamezp** [comment]
retest this please

**Comment by jamezp** [comment]
retest this please

File: logging/src/main/java/org/jboss/as/logging/handlers/SyslogHandlerResourceDefinition.java [hunk 1]
```diff
@@ -118,6 +120,11 @@ public void marshallAsElement(final AttributeDefinition attribute, final ModelNo
             .setValidator(EnumValidator.create(SyslogType.class, EnumSet.allOf(SyslogType.class)))
             .build();
 
+    public static final SimpleAttributeDefinition NAMED_FORMATTER = SimpleAttributeDefinitionBuilder.create(AbstractHandlerDefinition.NAMED_FORMATTER)
+            .setAlternatives()
```
Comment (bstansberry) [inline logging/src/main/java/org/jboss/as/logging/handlers/SyslogHandlerResourceDefinition.java]: Something missing?

Comment (bstansberry) [reply logging/src/main/java/org/jboss/as/logging/handlers/SyslogHandlerResourceDefinition.java]: Probably not; looks like cruft.

**Comment by bstansberry** [comment]
Minor point about the cruft aside this looks good.

File: logging/src/main/java/org/jboss/as/logging/handlers/SyslogHandlerResourceDefinition.java (hunk 1 above)
Comment (jamezp) [reply logging/src/main/java/org/jboss/as/logging/handlers/SyslogHandlerResourceDefinition.java]: This was to remove the alternatives as the `syslog-handler` doesn't and won't have a `formatter` attribute. All the other resources that have a `named-formatter` attribute have an alternative set of `formatter`. It probably makes sense to document why this is done though.

**Pull request merged**